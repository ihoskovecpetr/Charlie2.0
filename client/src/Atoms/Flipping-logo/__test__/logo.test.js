import React from "react";
import ReactDOM from "react-dom";
import FlippingLogo from "../logo";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";
import Logo from "../logo";
//import "jest-styled-components";

//import "@testing-library/jest-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<FlippingLogo></FlippingLogo>, div);
});

it("renders correctly", () => {
  const { getByTestId } = render(<FlippingLogo height={100} width={100} />);
  expect(getByTestId("logo")).toHaveClass("flip-card");
});

it("renders correctly", () => {
  const { getByTestId } = render(<FlippingLogo height={80} width={80} />);
  expect(getByTestId("logo")).toHaveClass("flip-card");
});

it("Matches snapshot", () => {
  const tree = renderer
    .create(<FlippingLogo height={80} width={80} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
