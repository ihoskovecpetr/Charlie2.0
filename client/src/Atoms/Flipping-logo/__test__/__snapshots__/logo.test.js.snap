// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="flip-card"
  data-testid="logo"
  style={
    Object {
      "height": 80,
      "width": 80,
    }
  }
>
  <div
    className="flip-card-inner"
    id="main_logo"
  >
    <div
      className="flip-card-front"
    >
      <img
        src="charlie.png"
      />
    </div>
    <div
      className="flip-card-back"
    >
      <img
        src="charlie.png"
      />
    </div>
  </div>
</div>
`;
